// slonik/dist/symbols.js
"use strict";

require("core-js/modules/es.symbol.description");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ValueListTokenSymbol = exports.UnnestTokenSymbol = exports.TupleTokenSymbol = exports.TupleListTokenSymbol = exports.SqlTokenSymbol = exports.RawSqlTokenSymbol = exports.IdentifierTokenSymbol = exports.IdentifierListTokenSymbol = exports.ComparisonPredicateTokenSymbol = exports.BooleanExpressionTokenSymbol = exports.AssignmentListTokenSymbol = exports.ArrayTokenSymbol = void 0;
const ArrayTokenSymbol = Symbol('SLONIK_ARRAY');
exports.ArrayTokenSymbol = ArrayTokenSymbol;
const AssignmentListTokenSymbol = Symbol('SLONIK_ASSIGNMENT_LIST');
exports.AssignmentListTokenSymbol = AssignmentListTokenSymbol;
const BooleanExpressionTokenSymbol = Symbol('SLONIK_BOOLEAN_EXPRESSION');
exports.BooleanExpressionTokenSymbol = BooleanExpressionTokenSymbol;
const ComparisonPredicateTokenSymbol = Symbol('SLONIK_COMPARISON_PREDICATE');
exports.ComparisonPredicateTokenSymbol = ComparisonPredicateTokenSymbol;
const IdentifierListTokenSymbol = Symbol('SLONIK_IDENTIFIER_LIST');
exports.IdentifierListTokenSymbol = IdentifierListTokenSymbol;
const IdentifierTokenSymbol = Symbol('SLONIK_IDENTIFIER');
exports.IdentifierTokenSymbol = IdentifierTokenSymbol;
const RawSqlTokenSymbol = Symbol('SLONIK_RAW_SQL');
exports.RawSqlTokenSymbol = RawSqlTokenSymbol;
const SqlTokenSymbol = Symbol('SLONIK_SQL');
exports.SqlTokenSymbol = SqlTokenSymbol;
const TupleListTokenSymbol = Symbol('SLONIK_TUPLE_LIST');
exports.TupleListTokenSymbol = TupleListTokenSymbol;
const TupleTokenSymbol = Symbol('SLONIK_TUPLE');
exports.TupleTokenSymbol = TupleTokenSymbol;
const UnnestTokenSymbol = Symbol('SLONIK_UNNEST');
exports.UnnestTokenSymbol = UnnestTokenSymbol;
const ValueListTokenSymbol = Symbol('SLONIK_VALUE_LIST');
exports.ValueListTokenSymbol = ValueListTokenSymbol;
